### PR TITLE
FR-1916  Email Respondent solicitor AssignedToJudge

### PR DESCRIPTION
**Pull request by ravinderpal**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FR-1916

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java [hunk 1]
```diff
@@ -46,11 +46,16 @@ public void sendConsentedHWFSuccessfulConfirmationEmail(CaseDetails caseDetails)
...
     }
 
-    public void sendAssignToJudgeConfirmationEmail(CaseDetails caseDetails) {
+    public void sendAssignToJudgeConfirmationEmailToApplicantSolicitor(CaseDetails caseDetails) {
         URI uri = buildUri(notificationServiceConfiguration.getAssignToJudge());
         sendNotificationEmail(notificationRequestMapper.createNotificationRequestForAppSolicitor(caseDetails), uri);
     }
 
+    public void sendAssignToJudgeConfirmationEmailToRespondentSolicitor(CaseDetails caseDetails) {
+        URI uri = buildUri(notificationServiceConfiguration.getAssignToJudge());
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:55]: see how we've done this for other methods, we have a base method that we call fro for app and resp specific methods

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java [hunk 2]
```diff
@@ -820,6 +820,46 @@ public void whenShouldNotSendContestedGeneralOrderEmailToRespondent_ThenTheEmail
...
     }
 
+    @Test
+    public void whenToggleEnabledAndShouldSendEmailToRespSolicitor_thenSendsEmail() {
+        when(featureToggleService.isRespondentSolicitorEmailNotificationEnabled()).thenReturn(true);
+        when(notificationService.shouldEmailRespondentSolicitor(any())).thenReturn(true);
+
+        notificationsController.sendAssignToJudgeConfirmationEmail(AUTH_TOKEN, buildCallbackRequest());
+
+        verify(notificationService).sendAssignToJudgeConfirmationEmailToRespondentSolicitor(any());
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java:830]: can you verify the caseDetails from buildCallbackRequest() is passed in this test

Comment (ravinderpal) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java:835]: done! 👍

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java (hunk 1 above)
Comment (ravinderpal) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:58]: done 👍

**Review by ravinderpal — COMMENTED** [review]
Changes requested applied

**Review by johnoneill46 — APPROVED** [review]

**Review by HarryH96 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java [hunk 3]
```diff
@@ -820,6 +823,51 @@ public void whenShouldNotSendContestedGeneralOrderEmailToRespondent_ThenTheEmail
...
+        final ArgumentCaptor<CaseDetails> requestCaptor = ArgumentCaptor.forClass(CaseDetails.class);
+
+        when(featureToggleService.isRespondentSolicitorEmailNotificationEnabled()).thenReturn(true);
+        when(notificationService.shouldEmailRespondentSolicitor(any())).thenReturn(true);
+
+        notificationsController.sendAssignToJudgeConfirmationEmail(AUTH_TOKEN, buildCallbackRequest());
+
+        verify(notificationService).sendAssignToJudgeConfirmationEmailToRespondentSolicitor(any());
+
+        verify(notificationService).sendAssignToJudgeConfirmationEmailToRespondentSolicitor(requestCaptor.capture());
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/NotificationsControllerTest.java:837]: you don't need to use an argument captor here (although I do love argument captors), it looks like you can just verify buildCallbackRequest().caseDetails is passed to this method call

**Pull request merged**